plugins/gcp/resoto_plugin_gcp/resources/compute.py

```diff
@@ -2715,18 +2721,48 @@ class GcpInstance(GcpResource, BaseInstance):
     instance_tags: Optional[GcpTags] = field(default=None)
 
     def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
-        # TODO: load custom machine types
-        pass
+        if self.instance_status == InstanceStatus.TERMINATED:
+            self._cleaned = True
+
+    def connect_machine_type(self, machine_type_link_ish: str, builder: GraphBuilder) -> None:
+        if not machine_type_link_ish.startswith("https://"):
+            machine_type_link_ish = (
+                f"https://www.googleapis.com/compute/v1/projects/{builder.project.id}/{machine_type_link_ish}"
+            )
+        machine_type = builder.node(clazz=GcpMachineType, link=machine_type_link_ish)
+        if machine_type:
+            self.instance_cores = machine_type.instance_cores
+            self.instance_memory = machine_type.instance_memory
+            self.instance_type = machine_type.name
+            builder.add_edge(from_node=self, reverse=True, node=machine_type)
 
     def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
         super().connect_in_graph(builder, source)
-        builder.add_edge(from_node=self, reverse=True, clazz=GcpMachineType, link=source["machineType"])
+        self.connect_machine_type(self.instance_machine_type, builder)
+
         for nic in self.instance_network_interfaces or []:
             builder.dependant_node(self, reverse=True, delete_same_as_default=True, clazz=GcpNetwork, link=nic.network)
             builder.dependant_node(
                 self, reverse=True, delete_same_as_default=True, clazz=GcpSubnetwork, link=nic.subnetwork
             )
 
+    @classmethod
+    def collect(cls: Type[GcpResource], raw: List[Json], builder: GraphBuilder) -> List[GcpResource]:
+        # Additional behavior: iterate over list of collected GcpInstances and for each:
+        # - extract machineType if custom
+        # - then create GcpMachineType object for each unique custom machine type
+        # - add the new objects to the graph
+        result = super().collect(raw, builder)
+        custom_machine_types = list(
+            set([instance.instance_machine_type for instance in result if "custom" in instance.instance_machine_type])
+        )
+        for machine_type in custom_machine_types:
+            zone = machine_type.split("/")[-3]
+            name = machine_type.split("/")[-1]
```

Comment:
a) you do not need to split twice - assign the variable and get the name and zone
b) can you add a comment with an example: hard to see if the access is correct